File: scm-java/src/main/scheme/primitives/PrimitiveConditionVariableBroadcastB.java
package scheme.primitives;
import scheme.*;

public class PrimitiveConditionVariableBroadcastB extends Primitive {
    @Override
    public String name() { return "condition-variable-broadcast!"; }

    @Override
    public String info() {
        return "Syntax: (condition-variable-broadcast! cv)\n" +
               "Library: (srfi 18)\n" +
               "Description: Broadcasts the condition variable, waking all waiting threads.\n" +
               "Example:\n" +
               "  (condition-variable-broadcast! cv)";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        SchemeConditionVariable cv = (SchemeConditionVariable) Value.asNativeValue(arguments[0]).value;
        cv.broadcast();
        return Value.NIL;
    }
}

File: scm-csharp/scheme/primitives/PrimitiveConditionVariableBroadcastB.cs
namespace scheme;

public class PrimitiveConditionVariableBroadcastB : Primitive
{
    public override string Name() => "condition-variable-broadcast!";

    public override string Info() =>
        "Syntax: (condition-variable-broadcast! cv)\n" +
        "Library: (srfi 18)\n" +
        "Description: Wakes up all threads waiting on the condition variable.\n" +
        "Example:\n" +
        "  (condition-variable-broadcast! cv)";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        SchemeConditionVariable cv = (SchemeConditionVariable) Value.AsNativeValue(arguments[0]).value;
        cv.Broadcast();
        return Value.NIL;
    }
}
